src/Microsoft.DotNet.Cli.Utils/NativeMethods.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class NativeMethods
+    {
+        internal static class Windows
+        {
+            internal enum JobObjectInfoClass : uint
+            {
+                JobObjectExtendedLimitInformation = 9,
+            }
+
+            [Flags]
+            internal enum JobObjectLimitFlags : uint
+            {
+                JobObjectLimitKillOnJobClose = 0x2000,
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct JobObjectBasicLimitInformation
+            {
+                public Int64 PerProcessUserTimeLimit;
+                public Int64 PerJobUserTimeLimit;
+                public JobObjectLimitFlags LimitFlags;
+                public UIntPtr MinimumWorkingSetSize;
+                public UIntPtr MaximumWorkingSetSize;
+                public UInt32 ActiveProcessLimit;
+                public UIntPtr Affinity;
+                public UInt32 PriorityClass;
+                public UInt32 SchedulingClass;
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct IoCounters
+            {
+                public UInt64 ReadOperationCount;
+                public UInt64 WriteOperationCount;
+                public UInt64 OtherOperationCount;
+                public UInt64 ReadTransferCount;
+                public UInt64 WriteTransferCount;
+                public UInt64 OtherTransferCount;
+            }
+
+            [StructLayout(LayoutKind.Sequential)]
+            internal struct JobObjectExtendedLimitInformation
+            {
+                public JobObjectBasicLimitInformation BasicLimitInformation;
+                public IoCounters IoInfo;
+                public UIntPtr ProcessMemoryLimit;
+                public UIntPtr JobMemoryLimit;
+                public UIntPtr PeakProcessMemoryUsed;
+                public UIntPtr PeakJobMemoryUsed;
+            }
+
+            [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+            internal static extern SafeWaitHandle CreateJobObjectW(IntPtr lpJobAttributes, string lpName);
+
+            [DllImport("kernel32.dll", SetLastError = true)]
+            internal static extern bool SetInformationJobObject(IntPtr hJob, JobObjectInfoClass jobObjectInformationClass, IntPtr lpJobObjectInformation, UInt32 cbJobObjectInformationLength);
+
+            [DllImport("kernel32.dll", SetLastError = true)]
+            internal static extern bool AssignProcessToJobObject(IntPtr hJob, IntPtr hProcess);
+        }
+
+        internal static class Posix
+        {
+            [DllImport("libc", SetLastError = true)]
+            internal static extern int kill(int pid, int sig);
+
+            internal const int SIGINT = 2;
+            internal const int SIGTERM = 15;
```

Comment:
Do we need to be concerned about these being defined differently in different implementations?

http://pubs.opengroup.org/onlinepubs/009695399/basedefs/signal.h.html

Doesn't appear to specify the values given to these macros.

Reply:
I don't think so given every platform we support uses these values.  I can't think of a single platform that uses different values (Solaris, *BSD, Linux, BeOS...).

Reply:
OK. Just asking because a bunch of other macros defined by POSIX are not constant across the platforms we support.

Reply:
For example, errno.h values vary.

But https://en.wikipedia.org/wiki/Signal_(IPC) has a "portable number" column for the numbers that can be passed to kill on the command line. 

> For most signals the corresponding signal number is implementation-defined. This column lists the numbers specified in the POSIX standard.

My (probably incorrect) pedantic reading so far is that one could theoretically have a POSIX compliant system where `kill -2` sent SIGINT but signal.h defined SIGINT != 2. That would be silly though. ;)

Reply:
FWIW, I'm sensitive to this because we got bitten a lot in the early days of x-plat corefx. I was on the corefx team at the time and led an effort to clean it up with the strategy I wrote down here: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md#unix-shims

Reply:
Yeah, luckily the POSIX code used here is well defined and has no struct layouts required.

Agreed that it can get hairy really quickly (not to mention now we have multiple libc implementations, e.g. musl).